sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

```diff
@@ -91,6 +91,20 @@ public bool TryGetProperty(string name, out MutableJsonElement value)
             return true;
         }
 
+        public int GetArrayLength()
+        {
+            EnsureValid();
+
+            EnsureArray();
+
+            if (Changes.TryGetChange(_path, _highWaterMark, out MutableJsonChange change))
+            {
+                return change.AsJsonElement().GetArrayLength();
```

Comment:
Out of curiosity, does this get cached? I would expect that any array usage would probably require numerous bindings. Or is the lookup expected to be O(1)?

Reply:
Great question, and I don't know offhand, but added to the list of perf follow-ups here: https://github.com/Azure/azure-sdk-for-net/issues/33856